SQL/database_changelog.txt

```diff
@@ -1,15 +1,541 @@
 Any time you make a change to the schema files, remember to increment the database schema version. Generally increment the minor number, major should be reserved for significant changes to the schema. Both values go up to 255.
 
-The latest database version is 6.2; The query to update the schema revision table is:
+The latest database version is 7.0; The query to update the schema revision table is:
 
-INSERT INTO `schema_revision` (`major`, `minor`) VALUES (6, 2);
+INSERT INTO `schema_revision` (`major`, `minor`) VALUES (7, 0);
 or
-INSERT INTO `SS13_schema_revision` (`major`, `minor`) VALUES (6, 2);
+INSERT INTO `SS13_schema_revision` (`major`, `minor`) VALUES (7, 0);
 
 In any query remember to add a prefix to the table names if you use one.
 
 -----------------------------------------------------
 
+-----------------------------------------------------
+
+Version 7.0, 24 July 2023, by itsmeow
+Datumized preferences (TGUI Prefs)
+DO NOT RUN WITHOUT TAKING A FULL BACKUP.
+DO NOT RUN MORE THAN ONCE.
+DO NOT RUN COLUMN REMOVALS UNTIL DATA IS VERIFIED.
+- Allows nulls for `ss13_characters` columns (required for partial INSERT INTO ON DUPLICATE KEY UPDATE)
+- Alters some datatypes to fit new data.
+- Turns `ss13_characters`.`features` and `ss13_characters`.`custom_names` JSON objects into individual columns, removes the original columns.
+- Updates `ss13_preferences`.`preference_tag` to text.
+- Updates various `ss13_preferences` values to new ones.
+- Flips `ss13_preferences`.`key_bindings` JSON structure.
+
```

Comment:
This is too much for a simple CL entry. Put it in its own SQL file similar to the other ones. Call it something like `prefs_migration_8776.sql`